core/templates/pages/exploration-player-page/learner-experience/conversation-skin.component.ts

```diff
@@ -1241,22 +1282,36 @@ export class ConversationSkinComponent {
             this.explorationActuallyStarted = true;
           }
         }
-        if (!this.explorationPlayerStateService.isInQuestionMode()) {
+        if (
+          this.explorationPlayerStateService.isPresentingIsolatedQuestions()
+        ) {
           this.explorationPlayerStateService.onPlayerStateChange.emit(
             nextCard.getStateName());
-        } else {
+        } else if (
+          this.explorationPlayerStateService.isInQuestionPlayerMode()
```

Comment:
If question player mode implies that it's presenting isolated questions, then how can this branch ever execute (given the previous if clause)?

Reply:
Updated